src/pmi/simple/simple_pmi.c

```diff
@@ -201,6 +201,18 @@ int PMI_Init(int *spawned)
 #endif
 
     PMII_getmaxes(&PMI_kvsname_max, &PMI_keylen_max, &PMI_vallen_max);
+    /* we need construct a cmd like "cmd=put kvsname=%s key=%s value=%s\n",
+     * make sure it fits in PMIU_MAXLINE.
+     */
+    if (PMI_kvsname_max + PMI_keylen_max + PMI_vallen_max + 30 > PMIU_MAXLINE) {
+        if (PMI_keylen_max > 256) {
+            PMI_keylen_max = 256;
+            PMI_vallen_max = PMIU_MAXLINE - PMI_kvsname_max - PMI_keylen_max - 30;
+        } else {
+            PMI_vallen_max = PMIU_MAXLINE - PMI_kvsname_max - PMI_keylen_max - 30;
```

Comment:
this line is the same as in the `if` branch. can we remove the else and place this line outside the conditional?

Reply:
You are right. Will change.

Reply:
I suppose we could integrate `MPL_MAX` in here, too, but that's just nitpicking.

Reply:
It is 
```
if (PMI_keylen_max > 256) {
    PMI_keylen_max = 256;
}
```
versus
```
PMI_keylen_max = MPL_MIN(PMI_keylen_max, 256);
```
I am finding the `if-` version reads easier in this case.